### PR TITLE
[MINOR] add download link for 21.10.1

### PR DESCRIPTION
**Pull request by macfarla**
Signed-off-by: Sally MacFarlane <sally.macfarlane@consensys.net>

added download link and hash for 21.10.1

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -30,6 +30,10 @@
 ### Early Access Features
 - Enable plugins to expose custom JSON-RPC / WebSocket methods [#1317](https://github.com/hyperledger/besu/issues/1317)
 
+### Download Link
+https://hyperledger.jfrog.io/ui/native/besu-binaries/besu/21.10.1/besu-21.10.1.zip \
+SHA256: 877148a1a13a1065151cf8fb3fbb56ea1d01e3a942fa7ebd6bd4ea4d15101fee
```
Comment (garyschulte) [inline CHANGELOG.md:35]: I am getting a different SHA... 
shasum -a 256 ~/Downloads/besu-21.10.1.zip
064df81be49c31ebfc0e1a5f03357e8f6f9a7f5e7bcb82e2b187e2d58e1704eb

**Review by shemnon — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (shemnon) [reply CHANGELOG.md:35]: Gary's matches the homebrew hash - https://github.com/hyperledger/homebrew-besu/blob/main/besu.rb#L6

File: CHANGELOG.md [hunk 2]
```diff
@@ -30,6 +30,10 @@
 ### Early Access Features
 - Enable plugins to expose custom JSON-RPC / WebSocket methods [#1317](https://github.com/hyperledger/besu/issues/1317)
 
+### Download Link
+https://hyperledger.jfrog.io/ui/native/besu-binaries/besu/21.10.1/besu-21.10.1.zip \
```
Comment (shemnon) [inline CHANGELOG.md:34]: Also, not the typical DL link
```suggestion
https://hyperledger.jfrog.io/artifactory/besu-binaries/besu/21.10.1/besu-21.10.1.zip \
```

File: CHANGELOG.md (hunk 1 above)
Comment (macfarla) [reply CHANGELOG.md]: I had downloaded the native zip. fixed.

File: CHANGELOG.md (hunk 2 above)
Comment (macfarla) [reply CHANGELOG.md]: native!

File: CHANGELOG.md (hunk 1 above)
Comment (macfarla) [reply CHANGELOG.md]: yep I had the wrong link. fixed

**Review by pinges — APPROVED** [review]
LGTM

**Pull request merged**